graphql_compiler/macros/__init__.py

```diff
@@ -167,16 +171,25 @@ def get_schema_for_macro_definition(schema):
     """Returns a schema with macro directives.
 
     This returned schema can be used to validate macro definitions, and support GraphQL
-    macro editors, enabling them to autocomplete one the @macro_edge_definition and
-    @macro_edge_target directives.
+    macro editors, enabling them to autocomplete on the @macro_edge_definition and
+    @macro_edge_target directives.*
 
     Args:
         schema: GraphQLSchema over which we want to write macros
 
     Returns:
         GraphQLSchema usable for writing macros
     """
-    raise NotImplementedError()  # TODO(bojanserafimov): Implement
+
+    macro_definition_schema = copy(schema)
+    macro_definition_schema_directives_set = set(chain(macro_definition_schema.get_directives(),
```

Comment:
I would replace all directives with the usual ones, instead of keeping the ones defined in this schema. The macro system works only with our usual directives, and makes the extra effort to make sure that all the folds, tags and recurses work as expected. If this schema has anything other than the usual directives, then we can't promise the macro system will work at all.

If you do this, also make sure to either document that all directives are replaced, or add a precondition and assertion that this schema uses the usual directives.

Reply:
I'm thinking of asserting that all directives are the default rather than adding a precondition. Is this a performance concern?

Reply:
nope, go for it. It's only a concern if this function is called often, but it shouldn't.